File: csharp/bob/Bob.cs
using System.Linq;

public class Bob
{
    public string Hey(string input)
    {
        if (Bob.IsSilence(input))
        {
            return "Fine. Be that way!";
        }

        if (Bob.IsYelling(input))
        {
            return "Woah, chill out!";
        }

        if (Bob.IsQuestion(input))
        {
            return "Sure.";
        }

        return "Whatever.";
    }

    private static bool IsSilence(string input)
    {
        return string.IsNullOrWhiteSpace(input);
    }

    private static bool IsYelling(string input)
    {
        return input.Any(x => char.IsLetter(x)) && input.All(x => !char.IsLetter(x) || char.IsUpper(x));
    }

    private static bool IsQuestion(string input)
    {
        return input.EndsWith("?");
    }
}

File: javascript/bob/bob.js
var Bob = function() {};

Bob.prototype.hey = function(input) {
    if (this.isSilence(input)) {
        return "Fine. Be that way!";
    }

    if (this.isYelling(input)) {
        return "Woah, chill out!";
    }

    if (this.isQuestion(input)) {
        return "Sure.";
    }

    return "Whatever.";
};

Bob.prototype.isYelling = function (input) {
    return input.toUpperCase() === input && /[A-Z]+/.test(input);
}

Bob.prototype.isSilence = function (input) {
    return /^\s*$/.test(input);
}

Bob.prototype.isQuestion = function (input) {
    return input.indexOf("?") === input.length - 1;
}

module.exports = Bob;
